src/SwedbankPay/PathResolver.php

```diff
@@ -0,0 +1,157 @@
+  */
+class PathResolver
+{
+    /**
+      * Resolves the path.
+      *
+      * @param string $path The original path.
+      * @return string The resolved path (it might not exist).
+      */
+      public function resolve($path)
+      {
+          $isUnixPath = $this->isUnixPath($path);
+          $path = $this->absolutizeStart($path);
+          $path = $this->enforceSeparator($path);
+          $path = $this->absolutizeStart($path);
+
+          $segments = explode(DIRECTORY_SEPARATOR, $path);
+          $parts = array_filter($segments, 'strlen');
+          $absolutes = array();
+
+          foreach ($parts as $part) {
+              if ('.'  == $part) {
+                  continue;
+              }
+
+              if ('..' == $part) {
+                  array_pop($absolutes);
+              } else {
+                  $absolutes[] = $part;
+              }
+          }
+
+          $path = implode(DIRECTORY_SEPARATOR, $absolutes);
+
+          $path = $this->resolveSymlinks($path);
+          $path = !$isUnixPath ? '/' . $path : $path;
+
+          return $path;
+      }
+
+
+    /**
+      * Attempts to detect if path is relative; in which case, prepend the
+      * current working directory.
+      *
+      * @param string $path The path to make absolute.
+      * @return string The absolute path.
+      */
+    private function absolutizeStart($path) : string
+    {
+        $isUnixPath = $this->isUnixPath($path);
+
+        if (strpos($path, ':') === false && $isUnixPath) {
+            return getcwd() . DIRECTORY_SEPARATOR . $path;
+        }
+
+        return $path;
+    }
+
+
+    /**
+      * Enforce DIRECTORY_SEPARATOR as a directory separator.
+      *
+      * @param string $path The path to enforce.
+      * @return string The enforced path.
+      */
+    private function enforceSeparator($path) : string
+    {
+        return str_replace(array('/', '\\'), DIRECTORY_SEPARATOR, $path);
+    }
+
+
+    /**
+      * Makes the directories (parts) of the path absolute.
+      *
+      * @param string $path The path to make absolute.
+      * @return string The absolute path.
+      */
+    private function absolutizeParts($path) : string
+    {
+        $segments = explode(DIRECTORY_SEPARATOR, $path);
+        $parts = array_filter($segments, 'strlen');
+        $absolutes = array();
+
+        foreach ($parts as $part) {
+            if ('.'  == $part) {
+                continue;
+            }
+
+            if ('..' == $part) {
+                array_pop($absolutes);
+            } else {
+                $absolutes[] = $part;
+            }
+        }
+
+        return implode(DIRECTORY_SEPARATOR, $absolutes);
+    }
+
+
+    /**
+     * Resolve any symbolic links in the path.
+     *
+     * @param string $path The in which to resolve symbolic links.
+     * @return string The resolved path.
+     */
+    private function resolveSymlinks($path) : string
+    {
+        if (file_exists($path) && linkinfo($path) > 0) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The use of function linkinfo() is discouraged](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-php/pullRequest?prid=5303907)